lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -249,6 +252,7 @@ def __configure_port_routing(self, ajs):
         service = Service(self._pykube_api, k8s_service_obj)
         service.create()
         ingress = Ingress(self._pykube_api, k8s_ingress_obj)
+        ingress.version = self.runner_params.get("k8s_ingress_api_version")
```

Comment:
Try using
```suggestion
        ingress.version = self.runner_params["k8s_ingress_api_version"]
```
That should cause the default to be returned I think. Runner_params is  custom class that overrides subscripting behaviour.